wire-ios/Wire-iOS/Sources/UserInterface/Conversation/Content/MessageActions/MessageActionsViewController.swift

```diff
@@ -0,0 +1,111 @@
+//
+// Wire
+// Copyright (C) 2023 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import UIKit
+import WireDataModel
+
+class MessageActionsViewController: UIAlertController {
+    private static let MessageLabelMarker = "__CUSTOM_CONTENT_MARKER__"
+
+    static func controller(withActions actions: [MessageAction],
```

Comment:
Why don't you want to make an `init()` instead of a static method?

Reply:
can't set preferredStyle in custom init. And can't call super.init(title, message, preferredStyle) from that init